recipes/libavif/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_decoder": ["aom", "dav1d"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_decoder": "dav1d",
+    }
+    generators = "cmake", "cmake_find_package_multi"
+    exports_sources = "CMakeLists.txt", "patches/*"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    @property
+    def _has_dav1d(self):
+        return self.options.with_decoder == "dav1d"
+
+    def requirements(self):
+        self.requires("libaom-av1/3.1.2")
+        self.requires("libyuv/cci.20201106")
+        if self._has_dav1d:
+            self.requires("dav1d/0.9.1")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        root = self._source_subfolder
+        get_args = self.conan_data["sources"][self.version]
+        tools.get(**get_args, destination=root, strip_root=True)
+
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["AVIF_CODEC_AOM"] = True
+        if self._has_dav1d:
+            cmake.definitions["AVIF_CODEC_DAV1D"] = True
+            cmake.definitions["AVIF_CODEC_AOM_DECODE"] = False
+        cmake.configure()
+        return cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        self._configure_cmake().build()
+
+    @property
+    def _alias_path(self):
+        return os.path.join("lib", "conan-official-avif-targets.cmake")
+
+    def package(self):
+        self.copy("LICENSE", "licenses", self._source_subfolder)
+        self._configure_cmake().install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+        # TODO: remove in conan v2
+        alias = os.path.join(self.package_folder, self._alias_path)
+        content = textwrap.dedent("""\
+            if(TARGET avif::avif AND NOT TARGET avif)
+                add_library(avif INTERFACE IMPORTED)
+                set_property(
+                    TARGET avif PROPERTY
+                    INTERFACE_LINK_LIBRARIES avif::avif
+                )
+            endif()
+        """)
+        tools.save(alias, content)
+
+    def package_info(self):
+        self.cpp_info.requires = ["libyuv::libyuv", "libaom-av1::libaom-av1"]
+        if self._has_dav1d:
+            self.cpp_info.requires.append("dav1d::dav1d")
+
+        self.cpp_info.libs = ["avif"]
+        if self.options.shared:
+            self.cpp_info.defines = ["AVIF_DLL"]
```

Comment:
<avif/avif.h> has the following comment:

```
// AVIF_DLL should be defined if libavif is a shared library. If you are using
// libavif as CMake dependency, through CMake package config file or through
// pkg-config, this is defined automatically.
```

Since you seem to be defining `AVIF_DLL` manually, does this mean you are not using libavif as CMake dependency or through pkg-config?